src/models/state.d.ts

```diff
@@ -0,0 +1,168 @@
+export interface RankDetails {
+  slug: string;
+  title: string;
+  demographic: string;
+  content_rating: string;
+  genres: number[];
+  last_chapter: number;
+  md_covers: MdCovers;
+}
+
+export interface MdCovers {
+  vol?: string;
+  w: number;
+  h: number;
+  b2key: string;
+}
+
+type MangaApiResponse = { rank: RankDetails[] };
+
+export interface FirstChap {
+  chap: string;
+  hid: string;
+  lang: string;
+  group_name: string[];
+  vol: string;
+}
+
+export interface Links {
+  al: string;
+  ap: string;
+  bw: string;
+  kt: string;
+  mu: string;
+  amz: string;
+  cdj: string;
+  ebj: string;
+  mal: string;
+  raw: string;
+  engtl: string;
+}
+
+export interface Relates {
+  title: string;
+  slug: string;
+  hid: string;
+  md_covers: MdCovers;
+}
+
+export interface Recommendations {
```

Comment:
Consider making the interface names singular and make the properties of that type plural (if they are arrays)

i.e.
` export interface Recommendation {...} ` then ` recommendations : Recommendation[] `